src/Util.php

```diff
@@ -221,7 +221,7 @@ public static function getParamName($class)
 
         $parts = \explode('\\', $class);
         $last = \array_pop($parts);
-        $last = \preg_replace('/Query$/', '', $last); // for BoolQuery
+        $last = \preg_replace('/(Query|_)$/', '', $last); // for BoolQuery and Match_
```

Comment:
Replaced `_` suffix here, but the method can be overriden to return `match` instead. WDYT @ruflin ?

Reply:
Not sure I follow? If we go with `Query` suffix instead of `_`, will it just work?

Reply:
Yes